src/components/postgresConnect.ts

```diff
@@ -1,45 +1,231 @@
+      if (!payload) {
+        console.log(`Notification ${channel} received with no payload!`)
+        return
+      }
+      switch (channel) {
+        case 'trigger_notifications':
+          processTrigger(JSON.parse(payload))
+          break
+        case 'action_notifications':
+          executeAction(JSON.parse(payload), actionLibrary)
+          break
+      }
+    })
+  }
+
+  private getValuesPlaceholders = (object: { [key: string]: any }) =>
+    Object.keys(object).map((key, index) => `$${index + 1}`)
+
+  public static get Instance() {
+    return this._instance || (this._instance = new this())
+  }
+
+  public query = async (text: string, payload: any[]): Promise<QueryResult> => {
+    const client = await this.pool.connect()
+    try {
+      return await client.query(text, payload)
+    } finally {
+      // Make sure to release the client before any error handling,
+      // just in case the error handling itself throws an error.
+      client.release()
+    }
+  }
+
+  public addActionQueue = async (action: ActionQueuePayload): Promise<boolean> => {
+    try {
+      const result = await this.query(
+        `INSERT into action_queue (${Object.keys(
+          action
+        )}, time_queued) VALUES (${this.getValuesPlaceholders(action)}, CURRENT_TIMESTAMP)`,
+        Object.values(action)
+      )
+      return true
+    } catch (err) {
+      console.log(err.stack)
+      return false
+    }
+  }
+
+  public executeActionQueued = async (payload: ActionQueueExecutePayload): Promise<boolean> => {
+    try {
+      await this.query(
+        'UPDATE action_queue SET status = $1, error_log = $2, execution_time = CURRENT_TIMESTAMP WHERE id = $3',
+        Object.values(payload)
+      )
+      return true
+    } catch (err) {
+      console.log(err.stack)
+      return false
+    }
+  }
+
+  public getActionsQueued = async (
+    payload: ActionQueueGetPayload = { status: 'Scheduled' }
+  ): Promise<ActionQueue[]> => {
+    try {
+      const result = await this.query(
+        'SELECT id, action_code, parameters, execution_time FROM action_queue WHERE status = $1 ORDER BY execution_time',
+        Object.values(payload)
+      )
+      return result.rows as ActionQueue[]
+    } catch (err) {
+      console.log(err.stack)
+      return []
+    }
+  }
 
-pgClient.on('notification', (msg: DatabaseRecord) => {
-  switch (msg.channel) {
-    case 'trigger_notifications':
-      processTrigger(pgClient, JSON.parse(msg.payload))
-      break
-    case 'action_notifications':
-      executeAction(pgClient, JSON.parse(msg.payload), actionLibrary)
-      break
+  public addFile = async (payload: FilePayload): Promise<number> => {
+    try {
+      const result = await this.query(
+        `INSERT INTO file (${Object.keys(payload)}) VALUES (${this.getValuesPlaceholders(
+          payload
+        )}) RETURNING id`,
+        Object.values(payload)
+      )
+      return result.rows[0]
```

Comment:
I think this is what's causing the problem with file upload. (See my comment on main thread for test result)

The `registerFileInDB` method (filehandler.ts) is expecting a `number` to be returned, but its actually receiving an object with an `id` field. My original code was returning `result.rows[0].id`, so it'll need to return that value instead.

Reply:
Ah! Yes. That will be definitely a problem. Fixing now!